pilot/pkg/model/envoyfilter.go

```diff
@@ -40,8 +41,17 @@ type EnvoyFilterConfigPatchWrapper struct {
 	Operation networking.EnvoyFilter_Patch_Operation
 	// Pre-compile the regex from proxy version match in the match
 	ProxyVersionRegex *regexp.Regexp
+	// ProxyPrefixMatch provides a prefix match for the proxy version. The current API only allows
+	// regex match, but as an optimization we can reduce this to a prefix match for common cases.
+	// If this is set, ProxyVersionRegex is ignored.
+	ProxyPrefixMatch string
 }
 
+// versionPrefixMatch is a regex that matches a regex
+// Specifically, we are looking for a regex like `^1\.6.*`, which is what is used by telemetry v2
+// This is intended only as a performance optimization
+var versionPrefixMatch = regexp.MustCompile(`\^1\\.(?P<ver>.)\.\*`)
```

Comment:
nit: you might want to use something more restrictive than `.` in the inner match. Maybe `[0-9a-z-]`? Otherwise this can backfire if the inner match turns out to be a regex.

Reply:
I replaced this with well known list